test/test-readability.js

```diff
@@ -47,11 +48,44 @@ var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
   };
 });
 
+function suite(result, expectedContent, expectedMetadata) {
+  it("should return a result object", function() {
+    expect(result).to.include.keys("content", "title", "excerpt", "byline");
+  });
+
+  it("should extract expected content with jsdom", function() {
+    expect(expectedContent).eql(prettyPrint(result.content));
+  });
+
+  it("should extract expected title", function() {
+    expect(expectedMetadata.title).eql(result.title);
+  });
+
+  it("should extract expected byline", function() {
+    expect(expectedMetadata.byline).eql(result.byline);
+  });
+
+  it("should extract expected excerpt", function() {
+    expect(expectedMetadata.excerpt).eql(result.excerpt);
+  });
+}
+
+function removeCommentNodesRecursively(node) {
+  for (var n = 0, length = node.childNodes.length; n < length; n++) {
+    var child = node.childNodes[n];
+    if (!child) continue; // JSDOM is dumb sometimes :/
```

Comment:
I'm wrong about forgetting some elements because of the `n--` below, but I think the length explanation still stands.